x/relaystate.go

```diff
@@ -0,0 +1,47 @@
+package x
+
+import (
+	"net/http"
+
+	"github.com/gorilla/sessions"
+	"github.com/pkg/errors"
+)
+
+// SessionGetRelayState returns a string of the content of the relaystate for the current session.
+func SessionGetStringRelayState(r *http.Request, s sessions.StoreExact, id string, key interface{}) (string, error) {
```

Comment:
This looks quite complex - which means that it needs a test :)

Reply:
It's covered by the test in [continuity/manager_relaystate_test.go](https://github.com/ovh/kratos/blob/saml/continuity/manager_relaystate_test.go) :)